stable_baselines3/common/torch_layers.py

```diff
@@ -224,6 +228,76 @@ def forward(self, features: th.Tensor) -> Tuple[th.Tensor, th.Tensor]:
         return self.policy_net(shared_latent), self.value_net(shared_latent)
 
 
+class CombinedExtractor(BaseFeaturesExtractor):
+    """
+    Combined feature extractor for Dict observation spaces.
+    Builds a feature extractor for each key of the space
+
+    :param observation_space:
+    :param features_dim: Number of features extracted.
+        This corresponds to the number of unit for the last layer.
+    :param cnn_output_dim: Number of features to output from each cnn submodule
+    :param mlp_output_dim: Number of features to output from each mlp submodule
+    :param mlp_net_arch: Architecture of each mlp network module
+    :param activation_fn: The activation function to use within each mlp
+    :param comb_net_arch: Architecture of the combined network module which calculates the final feature extracted
+    """
+
+    def __init__(
+        self,
+        observation_space: gym.spaces.Dict,
+        features_dim: int = 64,
+        cnn_output_dim: int = 64,
+        mlp_output_dim: int = 64,
+        mlp_net_arch: List[int] = [64, 64],
+        activation_fn: Type[nn.Module] = nn.ReLU,
+        comb_net_arch: List[int] = [64, 64],
+    ):
+        super(CombinedExtractor, self).__init__(observation_space, features_dim=features_dim)
+
+        extractors = {}
+
+        total_concat_size = 0
+        for (key, subspace) in observation_space.spaces.items():
+
+            if is_image_space(subspace):
+                n_input_channels = subspace.shape[0]
+                cnn = nn.Sequential(
```

Comment:
This could be replaced with NatureCNN, no? This network seems small and I'd opt for something time-tested, and original DQN network has served people well so far.

Reply:
(After update) As in could this not use the NatureCNN extractor instead of creating the network separately here? I see it could save bit of duplicate code :). This could even be an argument to the extractor so one can change CNN extractor easier.